### PR TITLE
Apply knockback resistance from custom armor materials

### LINKED ISSUE #197: Allow custom armor materials to have knockback resistance
In Vanilla, the knockback resistance attribute is limited to Netherite armor (governed by a `material == ArmorMaterials.NETHERITE` check). This limitation should be removed.

### PR DESCRIPTION
**Pull request by williambl**
# Motivation
See #197. Armor materials other than Netherite don't get their knockback resistance applied in vanilla.

# Content

A mixin to let non-Netherite armor materials have knockback resistance, plus a testmod.

Closes #197.

### CODE REVIEW

File: library/item/item_extension/src/testmod/java/org/quiltmc/qsl/item/extension/test/ArmorKnockbackTest.java
Comment (EnnuiL) [inline library/item/item_extension/src/testmod/java/org/quiltmc/qsl/item/extension/test/ArmorKnockbackTest.java]: Nitpick:
```suggestion
	public void onInitialize(ModContainer mod) { }
```

File: library/item/item_extension/src/testmod/java/org/quiltmc/qsl/item/extension/test/ArmorKnockbackTest.java
Comment (EnnuiL) [inline library/item/item_extension/src/testmod/java/org/quiltmc/qsl/item/extension/test/ArmorKnockbackTest.java]: Hm, I wonder if this should be renamed to "Knockback Resolution Armor" or "Anti-Knockback Armor"

**Review by EnnuiL — APPROVED** [review]
Looks good!

File: library/item/item_extension/src/testmod/java/org/quiltmc/qsl/item/extension/test/ArmorKnockbackTest.java
Comment (williambl) [reply library/item/item_extension/src/testmod/java/org/quiltmc/qsl/item/extension/test/ArmorKnockbackTest.java]: I don't really think it's worth it - none of the items in any of the item ext. testmod even have translation entries anyway :P

Comment (EnnuiL) [reply library/item/item_extension/src/testmod/java/org/quiltmc/qsl/item/extension/test/ArmorKnockbackTest.java]: oh well, that's fair enough

File: library/item/item_extension/src/main/java/org/quiltmc/qsl/item/extension/mixin/ArmorItemMixin.java
Comment (Leo40Git) [inline library/item/item_extension/src/main/java/org/quiltmc/qsl/item/extension/mixin/ArmorItemMixin.java]: I've had a thought: maybe have this return a dummy `ArmorMaterial` (or hell, even `null`) in case someone edits Netherite to have 0 knockback resistance (since if that happens, the check will still succeed since `original == ArmorMaterials.NETHERITE`)?

Comment (OroArmor) [reply library/item/item_extension/src/main/java/org/quiltmc/qsl/item/extension/mixin/ArmorItemMixin.java]: I don't think thats really an issue, since the check only gates adding the knockback resistance, which is already calculated from the actual material. I will say there might be some merit to resetting back to the original material after the `if` statement incase any mod injects afterward.

**Review by OroArmor — APPROVED** [review]

File: library/item/item_extension/src/main/java/org/quiltmc/qsl/item/extension/mixin/ArmorItemMixin.java
Comment (Leo40Git) [reply library/item/item_extension/src/main/java/org/quiltmc/qsl/item/extension/mixin/ArmorItemMixin.java]: ...wait, does `@ModifyVariable` actually edit the local variable for the rest of the method? I thought it acted like a `@Redirect`.

Comment (OroArmor) [reply library/item/item_extension/src/main/java/org/quiltmc/qsl/item/extension/mixin/ArmorItemMixin.java]: It does modify for the rest of the method.

**Comment by LambdAurora** [comment]
Putting this here so no one forgets (hopefully):

Once this is merged, replace `org.quiltmc.qsl.rendering.entity.test.mixin.ArmorItemMixin` from #166 with this.

**Review by LambdAurora — APPROVED** [review]

**Comment by TheGlitch76** [comment]
This PR needs to be updated to 1.19.4

**Comment by williambl** [comment]
rebased onto 1.19.4.

**Review by EnnuiL — APPROVED** [review]
I've gone ahead and finally addressed all the concerns with the PR myself using maintainer powers; I can finally say it has my go-ahead

**Review by OroArmor — APPROVED** [review]

**Comment by EnnuiL** [comment]
This PR has entered a Final Comment Period of 3 days!

**Review by LambdAurora — APPROVED** [review]

**Pull request merged**
